File: src/S4Analytics/Models/EventFeatureSet.cs
using GeoJSON.Net.Feature;

namespace S4Analytics.Models
{
    public class EventFeatureSet
    {
        public string eventType; // crash or violation
        public int? featureCount; // total number of matching features in current extent
        public Extent featureExtent; // extent of features in feature set
        public Extent queryExtent; // queried extent for feature set
        public bool isSample; // more points exist for current extent
        public int? sampleSize; // total number of sampled events (will hover around 10,000 if nonzero)
        public double? sampleMultiplier; // a count of sample points can be multiplied by this number to approximate the actual count
        public FeatureCollection featureCollection; // GeoJSON feature collection
    }
}

File: src/S4Analytics/ClientApp/app/event-analysis/shared/event-feature-set.ts
// tslint:disable-next-line
import { Point, FeatureCollection } from 'geojson';

export class EventFeatureSet {
    queryToken: string; // query token used to retrieve feature set
    eventType: string; // crash or violation
    featureCount: number; // total number of matching features in current extent
    featureExtent?: { minX: number, minY: number, maxX: number, maxY: number }; // extent of features in feature set
    isSample: boolean; // more points exist for current extent
    sampleSize?: number; // total number of sampled events (will hover around 10,000 if defined)
    sampleMultiplier?: number; // a count of sample points can be multiplied by this number to approximate the actual count
    featureCollection: FeatureCollection<Point>;

    constructor(queryToken: string, eventType: string) {
        // initialize an empty feature set
        this.queryToken = queryToken;
        this.eventType = eventType;
        this.featureCount = 0;
        this.isSample = false;
        this.featureCollection = {
            features: [],
            type: 'FeatureCollection'
        };
    }
}
